api/app/Models/PoolCandidateSearchRequest.php

```diff
@@ -54,6 +54,22 @@ class PoolCandidateSearchRequest extends Model
         'request_status_changed_at' => 'datetime',
     ];
 
+    /**
+     * Boot function for using with PoolCandidateSearchRequest
+     *
+     * @return void
+     */
+    protected static function boot()
+    {
+        /** @var \App\Models\User */
+        $user = Auth::user();
+
+        parent::boot();
+        static::creating(function ($query) use ($user) {
+            $query->user_id = $user ? $user->id : null;
```

Comment:
You could probably avoid this ternary with:
```suggestion
            $query->user_id = $user?->id;
```